networkx/algorithms/sampling.py

```diff
@@ -0,0 +1,278 @@
+#     samples::
+
+#         >>> import networkx as nx
+#         >>> G = nx.wheel_graph(10)
+#         >>> N = 10000
+#         >>> sample = list(nx.uniform_independent_node_sample(G, size=N))
+#         >>> nx.estimate_mean(sample, lambda v: v)
+#         ???
+
+#     Estimating the mean of the node labels on a graph given a large number of
+#     samples, weighted according to degree::
+
+#         >>> import networkx as nx
+#         >>> G = nx.wheel_graph(10)
+#         >>> N = 10000
+#         >>> sample = list(nx.uniform_independent_node_sample(G, size=N))
+#         >>> nx.estimate_mean(sample, values, weights=G.degree)
+#         ???
+
+#     """
+#     if weights is None:
+#         weights = lambda x: 1
+#     up = sum(values(v) / weights(v) for v in sample)
+#     down = sum(1 / weights(v) for v in sample)
+#     return up / down
+
+
+# def estimate_relative_size(sample, labeled, weights=None):
+#     """Estimates the fraction of labeled objects given a sample of the entire
+#     population.
+
+#     Parameters
+#     ----------
+#     sample : iterable
+#        Iterable of objects.
+
+#     labeled : Boolean function
+#        This function takes elements from ``sample`` as inputs and must return
+#        ``True`` if and only if the element is labeled.
+
+#     weights : function
+#        This function takes elements from ``sample`` as inputs and must return
+#        a :class:`float` representing the weight of the element.
+
+#     Returns
+#     -------
+#     int
+#        An estimate of the fraction of labeled elements.
+
+#     Examples
+#     --------
+#     Estimate the fraction of labeled elements in the wheel graph on ten
+#     vertices, where the first five nodes are labeled::
+
+#         >>> import networkx as nx
+#         >>> G = nx.wheel_graph(10)
+#         >>> is_labeled = lambda v: v in range(5)
+#         >>> estimate_relative_size(S, labeled=is_labeled, weights=G.degree)
+#         0.50679999999999814
+
+#     """
+#     values = lambda s: 1 if labeled(s) else 0
+#     return estimate_mean(sample, values, weights)
+
+
+def sample_with_replacements(population, size=-1, weights=None):
+    """Generates random items with replacement from a population.
+
+    This function is an iterator generator.
+
+    Parameters
+    ----------
+    population : list
+       List of elements from which to sample.
+
+    size : int
+       Desired number of samples. If this is ``-1``, the default, then the
+       generator never stops.
+
+    weights : function
+       This must be a function whose inputs are elements from ``population``
+       and whose outputs are :class:`float`s, representing the weight of that
+       element when sampling. A higher weight means the elements is more likely
+       to be sampled.
+
+    Returns
+    -------
+    iterator
+       An iterator over elements of ``population``.
+
+    References
+    ----------
+    .. [1]: Gjoka, Minas, Maciej Kurant, Carter T. Butts,
```

Comment:
MLA style, for example, inverts the name order of the first author. Search for the paper on Google Scholar, then click the "Cite" link to see some examples.